### ISSUE TITLE
[Accessibility]: In Code Monitors add/edit page, "Learn more" link should read better for screen readers

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#35730]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34195

### Problem description

https://sourcegraph.com/code-monitoring/new
"Learn more" is not a descriptive link

![CleanShot 2022-05-19 at 11 15 43](https://user-images.githubusercontent.com/206864/169371789-86251e73-881b-4d7a-8f87-16dbaf1de767.png)

### Expected behavior

Use visually hidden elements so the link reads "Learn more about code monitors" to screen readers

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by GitStart. I have assigned the label 'gitstart' and added the assignee 'gitstart-sourcegraph'.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.
